### ISSUE TITLE
Utility for seamless optimizer param groups creation

### ISSUE THREAD

**Issue by saleml** [saleml/torchgfn#72]
In `train_hypergrid.py`, for example
```
    # 3. Create the optimizer
    params = [
        {
            "params": [
                val
                for key, val in parametrization.parameters.items()
                if "logZ" not in key
            ],
            "lr": args.lr,
        }
    ]
    if "logZ.logZ" in parametrization.parameters:
        params.append(
            {
                "params": [parametrization.parameters["logZ.logZ"]],
                "lr": args.lr_Z,
            }
        )
```
should ideally be a one liner, using a utility function.

**Comment by saleml** [comment]
Should be done as part of a fix to #77

**Comment by saleml** [comment]
Done in #89
